operation/fhir-operation-validate/src/main/java/org/linuxforhealth/fhir/operation/validate/ValidateOperation.java

```diff
@@ -131,20 +123,75 @@ private OperationOutcome buildResourceValidOperationOutcome(List<Issue> issues)
     
     
     /**
-     * Check if a resource validation mode code is valid.
+     * This method does the following validations.
+     * 1. Validate if a resource validation mode code is valid.
+     * 2. Validate an interaction for a specified resource type.
+     * 3. Validate if persistence layer implementation supports update/create mode if mode = update/create.
+     * 4. Validate if modes update and delete are only be used when the operation is invoked at the resource instance level
      *
      * @param modeParameter resource validation mode code to be validated
+     * @param resourceHelper Resource operation provider for loading related Library resources
+     * @param resourceType a valid resource type string
+     * @param type the invocation type:  SYSTEM, RESOURCE_TYPE, INSTANCE
+     * @param persistence the FHIRPersistence implementation to facilitate the calls to the underlying db 
      * @throws FHIROperationException when the resource validation mode code is invalid
      */
-    private void validateModeParameter(Parameter modeParameter) throws FHIROperationException {
+    private void validateModeParameter(Parameter modeParameter, FHIRResourceHelpers resourceHelper, String resourceType, Type type, FHIRPersistence persistence) throws FHIROperationException {
         if (modeParameter != null && modeParameter.getValue() != null) {
+            ModeType modetype;
             try {
-                ModeType.from(modeParameter.getValue().as(Code.class).getValue());
+                // Validate if a resource validation mode code is valid
+                modetype =  ModeType.from(modeParameter.getValue().as(Code.class).getValue());
             } catch (IllegalArgumentException e) {
                 String msg = "'" + modeParameter.getValue().as(Code.class).getValue() + "' is not a valid resource validation mode";
-                throw buildExceptionWithIssue(msg, IssueType.VALUE);
+                throw buildExceptionWithIssue(IssueSeverity.ERROR, IssueType.INVALID, msg, null);
+            }
+            
+            // Validate an interaction for a specified resource type.
+            if (ModeType.CREATE.value().equals(modetype.value())  
+                    || ModeType.UPDATE.value().equals(modetype.value()) 
+                    || ModeType.DELETE.value().equals(modetype.value())) {
```

Comment:
you can use `==` with enums
```suggestion
            if (ModeType.CREATE == modetype)  
                    || ModeType.UPDATE == modetype) 
                    || ModeType.DELETE == modetype)) {
```

Reply:
this has been fixed.